joint_state_broadcaster/src/joint_state_broadcaster.cpp

```diff
@@ -58,13 +77,38 @@ JointStateBroadcaster::command_interface_configuration() const
 controller_interface::InterfaceConfiguration JointStateBroadcaster::state_interface_configuration()
 const
 {
-  return controller_interface::InterfaceConfiguration{controller_interface::
-    interface_configuration_type::ALL};
+  controller_interface::InterfaceConfiguration state_interfaces_config;
+
+  if (joints_.empty() || interfaces_.empty()) {
```

Comment:
for good self-documentation I'd suggest to wrap this around a bit and implement

```
bool JointStateBroadcaster::claimAllAvailableInterfaces(){ return joints_.empty() || interfaces_.empty(); }
```
and then change this code (and above using this condition) to
```suggestion
  if (claimAllAvailableInterfaces()) {...
```